Slack Notifications: Added a cron to clear error rate limiting files …

vedanshujain opened the pull request:
…that are created whenever an error happens.

vedanshujain committed:
Slack Notifications: Added a cron to clear error rate limiting files that are created whenever an error happens.

git-svn-id: https://meta.svn.wordpress.org/sites/trunk/wordcamp.org@8020 74240141-8908-4e6f-9713-ba540dce6ec7

vedanshujain committed:
Slack Notifications: Temporarily disable notifications to debug error limiting

git-svn-id: https://meta.svn.wordpress.org/sites/trunk/wordcamp.org@8021 74240141-8908-4e6f-9713-ba540dce6ec7

vedanshujain committed:
Slack Notifications: Temporarily disable notifications to debug error limiting. For reals.

git-svn-id: https://meta.svn.wordpress.org/sites/trunk/wordcamp.org@8022 74240141-8908-4e6f-9713-ba540dce6ec7

iandunn commented on public_html/wp-content/mu-plugins/wcorg-misc.php:
```diff
@@ -442,18 +442,19 @@ function wcorg_central_modify_export_caps( $primitive_caps, $meta_cap ) {
-	if ( ! file_exists( $error_dir ) ) {
-		mkdir( $error_dir );
+	if ( ! file_exists( ERROR_RATE_LIMITING_DIR ) ) {
+		mkdir( ERROR_RATE_LIMITING_DIR );
```
It's probably pretty safe to assume that `/tmp` exists, and that PHP wouldn't have the permissions to create it if it didn't, but in general it's a good habit to use `wp_mkdir_p`, since it's a bit safer. Not a big deal in this context, just FYI.

iandunn commented on public_html/wp-content/mu-plugins/wcorg-misc.php:
```diff
@@ -543,11 +544,31 @@ function send_fatal_to_slack() {
+	if ( ! init_error_handling() ) {
+		return;
+	}
+	foreach ( new DirectoryIterator( ERROR_RATE_LIMITING_DIR ) as $file_info ) {
```
It'd be nice to have a blank line here, because the two statements are logically distinct. I think it makes the function more readable, similar to how a page of text with paragraphs is much easier to read than one without.

iandunn commented on public_html/wp-content/mu-plugins/wcorg-misc.php:
```diff
@@ -543,11 +544,31 @@ function send_fatal_to_slack() {
+
+}
+add_action( 'clear_error_rate_limiting_files', 'handle_clear_error_rate_limiting_files' );
+if ( ! wp_next_scheduled( 'clear_error_rate_limiting_files' ) ) {
```
I think this is also a good place for a blank line.

iandunn reviewed (approved):
Left a few minor comments, but LGTM :)

vedanshujain committed:
Slack: Enable back error notifications for WordCamp servers

This was disabled in r8022 for debugging rate limiting. I have not yet found out why rate limiting was not working properly, but enabling it anyway in case we need it.


git-svn-id: https://meta.svn.wordpress.org/sites/trunk/wordcamp.org@8023 74240141-8908-4e6f-9713-ba540dce6ec7

vedanshujain committed:
WordCamp: Added some empty lines to make code more readable. 


git-svn-id: https://meta.svn.wordpress.org/sites/trunk/wordcamp.org@8024 74240141-8908-4e6f-9713-ba540dce6ec7

iandunn commented:
🎉 

Can you delete this branch, and any others that you've created that are no longer in use?

coreymckrill commented:
Would deleting the branches also delete the commit histories and pull requests? Might be nice to keep those around, at least for a while, since we're often squashing down to one commit on SVN...
